openfl/component/director/experiment.py

```diff
@@ -189,7 +191,7 @@ def finish_active(self) -> None:
         self.__active_experiment_name = None
 
     @asynccontextmanager
-    async def get_next_experiment(self):
+    async def get_next_experiment(self) -> AsyncGenerator[Experiment, None]:
```

Comment:
For sync generator was used `Iterator`, maybe for async should be used `AsyncIterator`